### PR TITLE
removing test init - singleton installation of rook

### LINKED ISSUE #818: tests: remove baseSetup and init() from tests
When running integration tests conditionally with `go test -run MyTest`, baseSetup.init() runs overtime and hangs if not everything is setup correctly.

`init()` is part of the entire test binary not just the suite. consider creating a single suite for "smoke" and moving the rook setup to setup suite and removing rook from suite teardown.

### PR DESCRIPTION
**Pull request by dangula**
installing and uninstalling rook at per suite.  fix #818

### CODE REVIEW

File: tests/block/create_block_test.go [hunk 1]
```diff
@@ -21,35 +21,58 @@ import (
...
+	rc             contracts.RestAPIOperator
 	initBlockCount int
 }
 
 func (s *BlockImageCreateSuite) SetupSuite() {
 
 	var err error
+	Env = objects.NewManifest()
+	// this is going to be cleaned up after #774
+	if kv := os.Getenv("KUBE_VERSION"); kv != "" {
```
Comment (travisn) [inline tests/block/create_block_test.go]: Could you have two helper methods, one for install and one for install? Then each suite would just have a 1-liner for install and 1-liner for uninstall. That would remove a lot of duplication with checking env vars and such.

Comment (dangula) [inline tests/block/create_block_test.go]: yes, let me do that

File: tests/block/create_block_test.go [hunk 2]
```diff
@@ -40,16 +44,21 @@ func TestBlockCreate(t *testing.T) {
...
+	rc             contracts.RestAPIOperator
 	initBlockCount int
 }
 
 func (s *BlockImageCreateSuite) SetupSuite() {
 
 	var err error
 
-	s.testClient, err = clients.CreateTestClient(tests.Platform)
+	rook, err := installer.NewK8sRookhelper()
```
Comment (travisn) [inline tests/block/create_block_test.go]: need to handle the err

File: tests/block/k8s/integration/k8sBlockStorageIntegration_test.go [hunk 3]
```diff
@@ -35,6 +36,11 @@ import (
 // Rook Block Storage integration test
 // Start MySql database that is using rook provisoned block storage.
 // Make sure database is functional
+var (
+	// Env var containing environment manifest
+	Env  objects.EnvironmentManifest
+	rook *installer.InstallHelper
```
Comment (travisn) [inline tests/block/k8s/integration/k8sBlockStorageIntegration_test.go]: could the `Env` be put inside the InstallHelper? Then there is only one context object

File: tests/block/k8s/create_block_test.go [hunk 4]
```diff
@@ -57,7 +60,12 @@ func (s *K8sBlockImageCreateSuite) SetupSuite() {
 
 	var err error
 
-	s.testClient, err = clients.CreateTestClient(tests.Platform)
+	rook, err := installer.NewK8sRookhelper()
```
Comment (travisn) [inline tests/block/k8s/create_block_test.go]: could you just call `NewK8sRookHelper` inside the `InstallRookOnK8s`? Seems like you're not using that `rook` context anywhere after this

File: tests/framework/installer/install_helper.go [hunk 5]
```diff
@@ -138,11 +137,19 @@ func (h *InstallHelper) createk8sRookCluster(k8sHelper *utils.K8sHelper, k8svers
...
 //InstallRookOnK8s installs rook on k8s
-func (h *InstallHelper) InstallRookOnK8s(k8sversion string) (err error) {
-	if h.isRookInstalled {
+func (h *InstallHelper) InstallRookOnK8s() (k8sversion string, err error) {
+	env := objects.NewManifest()
+	// this is going to be cleaned up after #774
+	if kv := os.Getenv("KUBE_VERSION"); kv != "" {
+		env.K8sVersion = kv
+	}
+	//flag used for local debuggin purpose, when rook is pre-installed
```
Comment (travisn) [inline tests/framework/installer/install_helper.go]: typo: `debuggin`

File: tests/framework/installer/install_helper.go [hunk 6]
```diff
@@ -138,11 +137,19 @@ func (h *InstallHelper) createk8sRookCluster(k8sHelper *utils.K8sHelper, k8svers
 }
 
 //InstallRookOnK8s installs rook on k8s
-func (h *InstallHelper) InstallRookOnK8s(k8sversion string) (err error) {
-	if h.isRookInstalled {
+func (h *InstallHelper) InstallRookOnK8s() (k8sversion string, err error) {
+	env := objects.NewManifest()
+	// this is going to be cleaned up after #774
```
Comment (travisn) [inline tests/framework/installer/install_helper.go]: i'd suggest not referring to issue numbers unless there is a very complex issue that needs to be tracked

File: tests/block/k8s/create_block_test.go (hunk 4 above)
Comment (dangula) [inline tests/block/k8s/create_block_test.go]: rook context is not used anywhere in this test, but I was thinking about other tests where you kill pods and have to install something again, 
I know are using oo patterns where they are not needed, we want to clean up that as part of #774,
Now that I think about it, I  feel like installer can be static.

File: tests/block/k8s/integration/k8sBlockStorageIntegration_test.go (hunk 3 above)
Comment (dangula) [inline tests/block/k8s/integration/k8sBlockStorageIntegration_test.go]: cleaned it up

File: tests/framework/installer/install_helper.go (hunk 6 above)
Comment (dangula) [inline tests/framework/installer/install_helper.go]: ok, removed the comments.

File: tests/framework/installer/install_helper.go (hunk 5 above)
Comment (dangula) [inline tests/framework/installer/install_helper.go]: fixed.

File: tests/block/create_block_test.go (hunk 2 above)
Comment (dangula) [inline tests/block/create_block_test.go]: added them.

File: tests/block/k8s/create_block_test.go (hunk 4 above)
Comment (travisn) [inline tests/block/k8s/create_block_test.go]: i do like the pattern of each suite creating its own context, then they could have different settings for their environment if needed

File: tests/block/k8s/integration/k8sBlockStorageIntegration_test.go [hunk 7]
```diff
@@ -57,7 +61,13 @@ func (s *K8sBlockEnd2EndIntegrationSuite) SetupSuite() {
 
 	var err error
 
-	s.testClient, err = clients.CreateTestClient(tests.Platform)
+	rook, err := installer.NewK8sRookhelper()
```
Comment (travisn) [inline tests/block/k8s/integration/k8sBlockStorageIntegration_test.go]: it looks like the global `rook` var will never be set. You're re-defining it here with this assignment. Does `TearDownSuite()` not give a null ref? I would suggest a more specific variable name than `rook` as well. What about `kubeContext`?

**Pull request merged**